.github/workflows/pipeline.yml

```diff
@@ -83,34 +83,39 @@ jobs:
         run: |
           npm run clean
           npm run build
-      - name: Publish changed packages
+      - name: Publish new versions of packages
         run: |
           root_path=$(pwd)
 
-          echo "→ Finding paths to packages whose package.json was changed in the last commit"
-          changed_package_json_paths=$(
-            git diff --name-only HEAD HEAD~ |
-            grep -E 'packages/[^/]+/package.json'
-          )
-          echo "${changed_package_json_paths}"
+          echo "→ Finding packages"
+          all_package_json_files=$( find packages -maxdepth 2 -name package.json )
+          echo "${all_package_json_files}"
 
-          echo "${changed_package_json_paths}" | while read -r changed_package_json_path
+          echo "${all_package_json_files}" | while read -r package_json_files
           do
-            changed_package_path=$(dirname "${changed_package_json_path}")
-            cd "${root_path}/${changed_package_path}"
-
+            package_dir=$(dirname "${package_json_files}")
+            
+            cd "${root_path}/${package_dir}"
             name=$(jq -r .name package.json)
+
+            latest_version=$(npm view $name version)
```

Comment:
nice!

Reply:
oh what happens if the package is not found? i.e. if we add a new sub-package, this could break publishing. not sure what the tidiest form to suggest… `latest_version=$(npm view $name version || echo missing)`

Reply:
I've gone with your example.. the worst that could happen if it fails for another reason (npm is down or whatever) is that the publishing will break and need to be rerun (I think it's idempotent)